### PR TITLE
Add more python bindings

### PR DESCRIPTION
**Pull request by agrayver**
Add missing binds for step-49 

Part of #9015

### CODE REVIEW

File: contrib/python-bindings/source/triangulation_wrapper.cc [hunk 1]
```diff
@@ -35,6 +35,43 @@ namespace python
...
+      std::vector<Point<spacedim>> vertices(n_vertices);
+      for (size_t i = 0; i < n_vertices; ++i)
+        {
+          boost::python::list vertex =
+            boost::python::extract<boost::python::list>(vertices_list[i]);
+          for (int d = 0; d < spacedim; ++d)
+            vertices[i][d] = boost::python::extract<double>(vertex[d]);
+        }
+
+      const size_t               n_cells = boost::python::len(cells_vertices);
```
Comment (Rombur) [inline contrib/python-bindings/source/triangulation_wrapper.cc]: Do you know why there is so many blank spaces? This looks strange.

File: contrib/python-bindings/tests/triangulation_wrapper.py [hunk 2]
```diff
@@ -140,6 +156,14 @@ def test_subdivided_material_hyper_rectangle(self):
...
 
+    def test_hyper_cube_with_cylindrical_hole(self):
+        for dim in self.restricted_dim:
+            triangulation = Triangulation(dim[0])
+            triangulation.generate_hyper_cube_with_cylindrical_hole(inner_radius = .25,
+				outer_radius = .5, L = .5, repetitions = 1, colorize = False)
+            n_cells = triangulation.n_active_cells()
+            self.assertEqual(n_cells, 8)
+
     def test_generate_chees(self):
```
Comment (Rombur) [inline contrib/python-bindings/tests/triangulation_wrapper.py]: Can you fix the typo here `test_generate_cheese`

File: contrib/python-bindings/include/triangulation_wrapper.h [hunk 3]
```diff
@@ -137,6 +145,20 @@ namespace python
       boost::python::list &material_id,
       const bool           colorize = false);
 
+    /**
+     * This function produces a square in the xy-plane with a cylindrical
+     * hole in the middle. The square and the circle are centered at the
+     * origin. In 3d, this geometry is extruded in z direction to the
+     * interval [0,L].
```
Comment (bangerth) [inline contrib/python-bindings/include/triangulation_wrapper.h]: Is there a way to use `@copydoc` to just copy the documentation of the function I assume you're calling internally?

File: contrib/python-bindings/source/export_triangulation.cc [hunk 4]
```diff
@@ -291,11 +302,24 @@ namespace python
 
 
 
+  const char generate_hyper_cube_with_cylindrical_hole_docstring[] =
+    "This function produces a square in the xy-plane with a cylindrical     \n"
+    "The square and the hole in the middle. circle are centered at the      \n"
```
Comment (bangerth) [inline contrib/python-bindings/source/export_triangulation.cc]: There's a grammar problem here :-)

File: contrib/python-bindings/source/triangulation_wrapper.cc (hunk 1 above)
Comment (agrayver) [reply contrib/python-bindings/source/triangulation_wrapper.cc]: No idea. `make indent` insists on this...

File: contrib/python-bindings/include/triangulation_wrapper.h (hunk 3 above)
Comment (agrayver) [reply contrib/python-bindings/include/triangulation_wrapper.h]: Like that?

Comment (agrayver) [reply contrib/python-bindings/include/triangulation_wrapper.h]: If it works here, should I replace it everywhere (in a separate PR)? It is rather annoying to copy-paste...

**Comment by agrayver** [comment]
Thanks both, I addressed your comments.

File: contrib/python-bindings/include/triangulation_wrapper.h (hunk 3 above)
Comment (bangerth) [reply contrib/python-bindings/include/triangulation_wrapper.h]: Yes, I think that would be great -- this way things don't go out of date.

Have you checked that it actually works?

(I wished we could do the same for the `_docstring` text...

File: contrib/python-bindings/source/triangulation_wrapper.cc (hunk 1 above)
Comment (masterleinad) [reply contrib/python-bindings/source/triangulation_wrapper.cc]: `clang-format` (with our settings) aligns subsequent variable declarations both at the type name and the variable name.

**Comment by agrayver** [comment]
I have added a couple more small binds here -- with them, I can fully reproduce step-49.

File: contrib/python-bindings/include/triangulation_wrapper.h (hunk 3 above)
Comment (agrayver) [reply contrib/python-bindings/include/triangulation_wrapper.h]: > Have you checked that it actually works?

Hm, how do I do that? The documentation for the python bindings is not compiled as far as I can see...

Comment (bangerth) [reply contrib/python-bindings/include/triangulation_wrapper.h]: Oh, I see. That's unfortunate. Let's just assume that it works, then.

**Comment by Rombur** [comment]
This looks good. The failing tester is because there is no memory left. If there are no more comment, I will merge this tomorrow.

**Pull request merged**